### PR TITLE
feat(files): compact single-child directory chains

### PR DESCRIPTION
**Pull request by mrmans0n**
## Summary

- Adds `compactChain(from:)` to `FilesTabView` — detects single-child directory chains at render time and merges them into one row (e.g. `src/main/java/com/example`)
- Updates `renderNode` to use the compact chain: displays joined path label, toggles all chain paths together on click, uses terminal node for badges/pills/children
- Chain stops gracefully at: 2+ children, file children, unloaded dirs, or submodules — these show as normal rows

## Test Plan

- [x] 10 unit tests covering all `compactChain` boundary conditions (`FilesTabCompactChainTests`)
- [x] Existing FilesTab, FileTreeBuilder tests all pass
- [ ] Manual: open a project with deep single-child dirs (e.g. Java/Kotlin src tree) and verify chains collapse

### CODE REVIEW

File: Alas/Sources/Right/FilesTabView.swift [hunk 1]
```diff
@@ -70,17 +70,19 @@ struct FilesTabView: View {
...
                     Button {
                         guard canExpand else { return }
                         if open {
-                            openPaths.remove(node.path)
+                            for p in chain.chainPaths { openPaths.remove(p) }
                         } else {
-                            openPaths.insert(node.path)
-                            onLoadChildren(node.path)
+                            for p in chain.chainPaths { openPaths.insert(p) }
+                            onLoadChildren(terminal.path)
```
[the remark this answers is not in the document]
Comment (mrmans0n) [reply Alas/Sources/Right/FilesTabView.swift]: Fixed in f3c389f — on open, we now call `onLoadChildren` for every path in `chainPaths` (not just the terminal), so ignored/excluded siblings under compacted prefix dirs are discovered when the row is expanded.

File: Alas/Sources/Right/FilesTabView.swift [hunk 2]
```diff
@@ -91,62 +95,62 @@ struct FilesTabView: View {
...
-                            if isOffGit(node) {
+                            if isOffGit(terminal) {
                                 ghostRail(depth: depth)
                             }
                         }
                         .contentShape(Rectangle())
-                        .background(node.path == revealPath ? theme.color("bg-hover") : Color.clear)
-                        .id(node.id)
+                        .background(terminal.path == revealPath ? theme.color("bg-hover") : Color.clear)
+                        .id("dir:\(terminal.path)")
```
[the remark this answers is not in the document]
Comment (mrmans0n) [reply Alas/Sources/Right/FilesTabView.swift]: Fixed in 091b384 — two changes: (1) the highlight check now uses `chain.chainPaths.contains(revealPath ?? "")` so any path in the chain triggers the background highlight; (2) zero-size overlay anchors are added for each intermediate chain path so `ScrollViewReader.scrollTo("dir:src")` resolves to the compacted row.

File: Alas/Sources/Right/FilesTabView.swift [hunk 3]
```diff
@@ -91,62 +95,67 @@ struct FilesTabView: View {
...
                         }
                     }
                 }
-                .task(id: loadTaskID(for: node, open: open)) {
-                    if shouldAutoLoadChildren(for: node, open: open) {
-                        onLoadChildren(node.path)
+                .task(id: loadTaskID(for: terminal, open: open)) {
+                    if shouldAutoLoadChildren(for: terminal, open: open) {
+                        onLoadChildren(terminal.path)
                     }
```
[the remark this answers is not in the document]
Comment (mrmans0n) [reply Alas/Sources/Right/FilesTabView.swift:159]: Fixed in e6b8131 — the auto-load `.task` now iterates `chain.chainPaths` the same way the button handler does, so all chain paths are reloaded on refresh when the row is already open.

File: Alas/Sources/Right/FilesTabView.swift [hunk 4]
```diff
@@ -91,62 +95,67 @@ struct FilesTabView: View {
...
                                 submodulePill
                             }
-                            if let badge = node.badge {
+                            if let badge = terminal.badge {
                                 StatusBadge(status: badge)
                             }
-                            if isOffGit(node) {
-                                visibilityPill(node)
+                            if isOffGit(terminal) {
+                                visibilityPill(terminal)
```
[the remark this answers is not in the document]
Comment (mrmans0n) [reply Alas/Sources/Right/FilesTabView.swift:116]: Fixed in 53ddea8 — added `children[0].visibility == current.visibility` to the `compactChain` guard, so the chain stops at any visibility boundary. An ignored dir will not compact with a tracked child (and vice versa), keeping off-git styling intact. Added two tests: one verifying the chain stops at a boundary, one verifying same-visibility chains (both ignored) still compact correctly.
